### PR TITLE
[EDU-728] - Limit notification clarifications

### PR DESCRIPTION
**Pull request by tbedford**
## Description

This PR adds some clarifications around limit notifications.

* [EDU-789](https://ably.atlassian.net/browse/EDU-789)
* [EDU-728](https://ably.atlassian.net/browse/EDU-728).

## Review

* [Limits page](https://ably-docs-edu-728-fix-l-rxi6u8.herokuapp.com/general/limits/)

### CODE REVIEW

File: content/general/limits.textile [hunk 1]
```diff
@@ -114,6 +130,33 @@ Message counts and rates are the number of "messages":/glossary#messages publish
...
+h4(#message-notifications). Message notifications
+
+Message limit notifications are split into hourly, monthly and message rate notifications. 
+
+h5(#message-hourly-notifications). Hourly notifications
+
+There are three types of hourly message limit notifications you may receive:
+
+* @messages.hourly.warning.count@ - when you reach 80% of your pre-paid quota for the number of messages.
+* @messages.hourly.soft.count@ - when your pre-paid quota for the number of messages has been exceeded.
```
Comment (m-hulbert) [inline content/general/limits.textile:142]: One minor point is that the descriptions for these 2 notifications aren't consistent with the others.

File: content/general/limits.textile [hunk 2]
```diff
@@ -415,14 +440,25 @@ API request limits are the maximum number of REST API requests that can be made
 
 If the API request rate is exceeded, a "global instantaneous rate limit":#global-rates will be applied to API requests. The hourly API request limit is a "time-based limit":#time, meaning no additional API requests will be accepted until the following hour if the hard limit is hit.
 
-h3(#api-warnings). API warnings
+h3(#api-requests-notifications). API limit notifications
+
+API requests limit notifications are split into hourly and message rate notifications.
```
Comment (m-hulbert) [inline content/general/limits.textile:445]:
```suggestion
API requests limit notifications are split into hourly and request rate notifications. 
```

File: content/general/limits.textile [hunk 3]
```diff
@@ -178,14 +241,25 @@ h3(#token-size). Token request size
 
 The token request size is the maximum size of a signed token request that will be accepted by the Ably platform.
 
-h3(#token-warnings). Token request warnings
+h3(#token-request-notifications). Token request notifications
+
+Token request limit notifications are split into hourly and message rate notifications.
```
Comment (m-hulbert) [inline content/general/limits.textile:246]:
```suggestion
Token request limit notifications are split into hourly and request rate notifications. 
```

**Review by m-hulbert — CHANGES_REQUESTED** [review]
A couple of minor points I noticed reading through and then LGTM 👍

File: content/general/limits.textile (hunk 1 above)
Comment (tbedford) [reply content/general/limits.textile:142]: Fixed.

**Review by m-hulbert — APPROVED** [review]

File: content/general/limits.textile [hunk 4]
```diff
@@ -114,6 +130,33 @@ Message counts and rates are the number of "messages":/glossary#messages publish
...
+h4(#message-notifications). Message notifications
+
+Message limit notifications are split into hourly, monthly and message rate notifications. 
+
+h5(#message-hourly-notifications). Hourly notifications
+
+There are three types of hourly message limit notifications you may receive:
+
+* @messages.hourly.warning.count@ - when you reach 80% of your pre-paid quota.
+* @messages.hourly.soft.count@ - when your pre-paid quota has been exceeded.
```
Comment (paddybyers) [inline content/general/limits.textile:142]: "when your pre-paid quota has been exceeded" -> "when the soft limit on the account, derived from the pre-paid quota, has been exceeded"

File: content/general/limits.textile [hunk 5]
```diff
@@ -114,6 +130,33 @@ Message counts and rates are the number of "messages":/glossary#messages publish
...
+
+* @messages.monthly.warning.count@ - when you reach 80% of your pre-paid quota.
+* @messages.monthly.soft.count@ - when your pre-paid quota has been exceeded.
+* @messages.monthly.hard.count@ - when the hard limit on the account has been exceeded.
+
+h5(#message-rate-notifications). Message rate notifications
+
+There are two types of message rate limit notifications you may receive:
+
+* @messages.maxRate.warning@ - when you reach 50% of your pre-paid quota.
```
Comment (paddybyers) [inline content/general/limits.textile:157]: when you reach 50% of the hard limit on the account

File: content/general/limits.textile [hunk 6]
```diff
@@ -124,6 +167,26 @@ Bandwidth is the amount of data transferred through "messages":/glossary#message
...
+h4(#bandwidth-notifications). Bandwidth notifications
+
+Bandwidth limit notifications are split into hourly and monthly notifications. 
+
+h5(#bandwidth-hourly-notifications). Hourly notifications
+
+There are three types of hourly bandwidth limit notifications you may receive:
+
+* @messages.hourly.warning.data@ - when you reach 80% of your pre-paid quota.
+* @messages.hourly.soft.data@ - when your pre-paid quota has been exceeded.
```
Comment (paddybyers) [inline content/general/limits.textile:179]: "when your pre-paid quota has been exceeded" -> "when the soft limit on the account, derived from the pre-paid quota, has been exceeded"

File: content/general/limits.textile [hunk 7]
```diff
@@ -178,14 +241,25 @@ h3(#token-size). Token request size
...
 
 * @tokenRequests.hourly.warning@ - when you reach 80% of your pre-paid quota.
 * @tokenRequests.hourly.soft@ - when your pre-paid quota has been exceeded.
 * @tokenRequests.hourly.hard@ - when the hard limit on the account has been exceeded.
 
+h4(#token-request-rate-notifications). Token request rate limit notifications
+
+There are two types of token request rate limit notifications you may receive:
+
+* @tokenRequests.maxRate.warning@ - when you reach 50% of your pre-paid quota.
```
Comment (paddybyers) [inline content/general/limits.textile:260]: when you reach 50% of the hard limit on the account

File: content/general/limits.textile [hunk 8]
```diff
@@ -415,14 +440,25 @@ API request limits are the maximum number of REST API requests that can be made
...
 
 * @apiRequests.hourly.warning@ - when you reach 80% of your pre-paid quota.
 * @apiRequests.hourly.soft@ - when your pre-paid quota has been exceeded.
 * @apiRequests.hourly.hard@ - when the hard limit on the account has been exceeded.
 
+h4(#api-requests-rate-notifications). API requests rate notifications
+
+There are two types of API requests rate limit notifications you may receive:
+
+* @apiRequests.maxRate.warning@ - when you reach 50% of your pre-paid quota.
```
Comment (paddybyers) [inline content/general/limits.textile:459]: when you reach 50% of the hard limit on the account

**Review by paddybyers — CHANGES_REQUESTED** [review]
I'm not convinced that describing the soft limits as "your pre-paid quota" is really correct in many of these cases.

A package contains a pre-paid quota for peak connections, peak channels, and monthly messages. Those package parameters are then translated into values for other limits - for example hourly messages. But it's not quite correct to describe the hourly allowance as an hourly "pre-paid quota", because the user didn't pay an amount for any given hour. We do define "soft limits" in the introduction on this page, so I think it would be more accurate to describe in that way where possible. I've added comments to that effect.

**Review by paddybyers — APPROVED** [review]
LGTM, thanks

**Pull request merged**